gateway/gateway.go

```diff
@@ -540,7 +544,7 @@ func (gateway *HandleT) getJobFromRequest(req *webRequestT) (job *jobsdb.JobT, n
 			return
 		}
 	}
-	numEvents = len(gjson.GetBytes(body, "batch").Array())
+	jobData.numEvents = len(gjson.GetBytes(body, "batch").Array())
```

Comment:
Should we consider using `gjson.GetMany` ([link](https://github.com/tidwall/gjson#get-multiple-values-at-once)) instead of decoding the json multiple times here and below for different keys ?

Reply:
I'm not entirely sure... the number of events in a batch could change and we'd have to dynamically create all the paths that we'd like to get from the batched payload.
Not that we couldn't do it, but it doesn't look as neat in my head.
I'm still thinking of a way to make this work, I'll post an update tomorrow.

Reply:
performance-wise it should be no different with what we are doing now.

Reply:
deferring this suggestion to a different PR in the future, includes some refactor and needs some more tests to be added.